### ISSUE TITLE
NextJS 15 `unmet peer react@"^16.11.0 || ^17.0.0 || ^18.0.0": found 19.0.0-rc-65a56d0e-20241020`

### ISSUE THREAD

**Issue by Yonom** [vercel/swr#3028]
# Bug

The supported React peer dep version range for v2.2.5 does not include React 19.0.0-rc
The `main` branch does, however, no new versions were released in the previous 5 months.

The dependency `use-sync-external-store` has a similar issue

```
│   └─┬ swr 2.2.5
│     ├── ✕ unmet peer react@"^16.11.0 || ^17.0.0 || ^18.0.0": found 19.0.0-rc-65a56d0e-20241020
│     └─┬ use-sync-external-store 1.2.2
│       └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0-rc-65a56d0e-20241020
```

swr 2.2.5
nextjs 15.0.0
react 19.0.0-rc-65a56d0e-20241020
using pnpm as package manager

**Comment by DevOsaWebsite** [comment]
@Yonom try **--force** option

**Comment by Yonom** [comment]
@promer94 not a "good first issue", the fix is already landed in main, it just needs to be deployed to npm

**Comment by TomerHeber** [comment]
Can this be deployed?
I'm "forced" to use `--force` with nextjs 15.

Thanks!

**Comment by aabughazaleh** [comment]
you can use this in package.json (make sure to use the react version you use in your project, my "19.0.0-rc-66855b96-20241106"):

"overrides": {
    "html-react-parser": {
      "react": "19.0.0-rc-66855b96-20241106"
    }
  },

**Comment by dgtllion** [comment]
> you can use this in package.json (make sure to use the react version you use in your project, my "19.0.0-rc-66855b96-20241106"):
> 
> ```
>   "overrides": {
>     "html-react-parser": {
>       "react": "19.0.0-rc-66855b96-20241106"
>     },
>     "swr": {
>       "react": "19.0.0-rc-66855b96-20241106"
>     }
>   },
> ```

ah, nice it worked for me. For `pnpm` i ended up with:
```
{
  "pnpm": {
    "overrides": {
      ...
      "swr": "npm:react@19.0.0-rc.1",
    },
  }
}
```

**Comment by dgtllion** [comment]
> you can use this in package.json (make sure to use the react version you use in your project, my "19.0.0-rc-66855b96-20241106"):
> 
> ```
>   "overrides": {
>     "html-react-parser": {
>       "react": "19.0.0-rc-66855b96-20241106"
>     },
>     "swr": {
>       "react": "19.0.0-rc-66855b96-20241106"
>     }
>   },
> ```

ah, nice it worked for me. For `pnpm` i ended up with:
```
{
  "pnpm": {
    "overrides": {
      ...
      "swr>react": "19.0.0-rc.1",
    },
  }
}
```

**Comment by dhedegaard-digisense** [comment]
When can we expect a new release, that's compatible with React 19?

**Comment by Katyi** [comment]
What version nextjs I should use for avoid this issue?

**Comment by aabughazaleh** [comment]
> What version nextjs I should use for avoid this issue?

you can use nextjs 14

**Comment by dgtllion** [comment]
now react 19 is out is it possible to update swr?

**Comment by koba04** [comment]
We've just released v2.3.0, which supports React 19, so I think this issue would be fixed.
https://github.com/vercel/swr/releases/tag/v2.3.0
